### DISCUSSION TITLE
How set JUST_CHOOSER for preview syntax highlighting?

### DISCUSSION THREAD

**Discussion by paidhi** [casey/just#2356] [Q&A]
I would like to pimp `just --choose` to get syntax highlighting in the preview.

`fzf` is installed on my system but I can't figure out how to set the JUST_CHOOSER environment variable in a way that a tool like `bat` or `highlight` is used in conjunction with `fzf` to show syntax coloring.

My configuration of `bat` [supports Justfile syntax](https://hynek.me/til/bat-justfile/).
If I pipe to `bat` or `highlight` manually it works. E.g.:
```
just --show myrecipe | bat -l just -pp

just --show myrecipe | highlight --out-format=ansi --syntax=makefile
```

Has anyone ever successfully done this and could give me a hint how to set `JUST_CHOOSER`?

**Comment by laniakea64** [comment]
What exact values of `JUST_CHOOSER` have you tried?
Are you invoking `bat` with `--color=always` option?

**Comment by jwdeane** [comment]
As noted above, the following works for me:

```sh
just --choose --chooser "fzf --multi --preview 'just --show {} | bat --color=always -l just -pp'"
```

My preference is to use a custom `choose` recipe. This gives me the benefit of including submodules, and allowing all recipes to be previewed/selected, not just those without arguments.

```sh
_choose:
    @just --summary | \
        tr ' ' '\n' | \
        sort -r | \
        fzf --multi --preview 'just --show {} | bat --color=always -l just -pp' | \
        xargs just
```
